src/plugins/discover/public/utils/get_sharing_data.ts

```diff
@@ -60,10 +60,20 @@ export async function getSharingData(
 
   return {
     getSearchSource: (absoluteTime?: boolean): SerializedSearchSourceFields => {
+      const absoluteFilter = data.query.timefilter.timefilter.createFilter(index);
       const timeFilter = absoluteTime
-        ? data.query.timefilter.timefilter.createFilter(index)
+        ? absoluteFilter
```

Comment:
Ok, I think I can suggest a way to omit any potential race condition:

So I'd suggest moving retrieving absolute and relative time filter to L64

```
const absoluteTimeFilter = data.query.timefilter.timefilter.createFilter(index);
const relativeTimeFilter = data.query.timefilter.timefilter.createRelativeFilter(index);
```

And then depending on `absoluteTime` use it like this on L70

```
const timeFilter = absoluteTime
        ? absoluteTimeFilter
        : relativeTimeFilter;
```

this would prevent the race condition when auto refresh is on and refreshing before `getSearchSource` is called .. because then absolute time filter would return a different result when called by `getSearchSource`